Library/Homebrew/livecheck.rb

```diff
@@ -109,38 +114,45 @@ def skip?
   # the `@strategy` instance variable when no argument is provided. The strategy
   # symbols use snake case (e.g. `:page_match`) and correspond to the strategy
   # file name.
-  #
-  # @param symbol [Symbol] symbol for the desired strategy
-  # @return [Symbol, nil]
-  def strategy(symbol = nil, &block)
+  sig {
+    params(
+      # Symbol for the desired strategy.
+      symbol: Symbol,
+      block:  T.nilable(Proc),
+    ).returns(T.nilable(Symbol))
+  }
+  def strategy(symbol = T.unsafe(nil), &block)
     @strategy_block = block if block
 
     case symbol
     when nil
       @strategy
     when Symbol
       @strategy = symbol
-    else
-      raise TypeError, "Livecheck#strategy expects a Symbol"
     end
   end
 
+  sig { returns(T.nilable(Proc)) }
   attr_reader :strategy_block
 
   # Sets the `@url` instance variable to the provided argument or returns the
   # `@url` instance variable when no argument is provided. The argument can be
   # a `String` (a URL) or a supported `Symbol` corresponding to a URL in the
   # formula/cask/resource (e.g. `:stable`, `:homepage`, `:head`, `:url`).
-  # @param val [String, Symbol] URL to check for version information
-  # @return [String, nil]
-  def url(val = nil)
-    case val
+  sig {
+    params(
+      # URL to check for version information.
+      url: T.any(String, Symbol),
+    ).returns(T.nilable(T.any(String, Symbol)))
+  }
+  def url(url = T.unsafe(nil))
+    case url
     when nil
       @url
     when String, :head, :homepage, :stable, :url
-      @url = val
-    else
-      raise TypeError, "Livecheck#url expects a String or valid Symbol"
+      @url = url
+    when Symbol
```

Comment:
Is there a reason that there is no longer an `else` catchall?

Reply:
We wanted to let `sorbet-runtime` handle this, no? It should be unreachable in that case.

Reply:
Agreed. I'm going to get https://github.com/Homebrew/brew/pull/15326 merged ASAP with an exception for `brew readall` until it's fixed.